pkg/blob/controllerserver.go

```diff
@@ -768,18 +777,62 @@ func (d *Driver) copyBlobContainer(_ context.Context, req *csi.CreateVolumeReque
 }
 
 // copyVolume copies a volume form volume or snapshot, snapshot is not supported now
-func (d *Driver) copyVolume(ctx context.Context, req *csi.CreateVolumeRequest, accountKey, dstContainerName, storageEndpointSuffix string) error {
+func (d *Driver) copyVolume(ctx context.Context, req *csi.CreateVolumeRequest, accountSASToken, dstContainerName, storageEndpointSuffix string) error {
 	vs := req.VolumeContentSource
 	switch vs.Type.(type) {
 	case *csi.VolumeContentSource_Snapshot:
 		return status.Errorf(codes.InvalidArgument, "copy volume from volumeSnapshot is not supported")
 	case *csi.VolumeContentSource_Volume:
-		return d.copyBlobContainer(ctx, req, accountKey, dstContainerName, storageEndpointSuffix)
+		return d.copyBlobContainer(ctx, req, accountSASToken, dstContainerName, storageEndpointSuffix)
 	default:
 		return status.Errorf(codes.InvalidArgument, "%v is not a proper volume source", vs)
 	}
 }
 
+func (d *Driver) authorizeAzcopyWithIdentity() ([]string, error) {
+	azureAuthConfig := d.cloud.Config.AzureAuthConfig
+	var authAzcopyEnv []string
+	if len(azureAuthConfig.AADClientSecret) > 0 {
+		klog.V(2).Infof("use service principal to authorize azcopy")
+		authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopyAutoLoginType+"=SPN"))
+		if azureAuthConfig.AADClientID == "" {
+			return []string{}, fmt.Errorf("AADClientID and AADClientSecret must be set when use service principal")
+		}
+		authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopySPAApplicationID+"="+azureAuthConfig.AADClientID))
+		authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopySPAClientSecret+"="+azureAuthConfig.AADClientSecret))
+		if azureAuthConfig.TenantID != "" {
+			authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopyTenantID+"="+azureAuthConfig.TenantID))
+			klog.V(2).Infof(fmt.Sprintf("set AZCOPY_TENANT_ID=%s successfully", azureAuthConfig.TenantID))
+		}
+		return authAzcopyEnv, nil
+	}
+	if azureAuthConfig.UseManagedIdentityExtension {
+		authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopyAutoLoginType+"=MSI"))
+		if len(azureAuthConfig.UserAssignedIdentityID) > 0 {
+			klog.V(2).Infof("use user assigned managed identity to authorize azcopy")
+			authAzcopyEnv = append(authAzcopyEnv, fmt.Sprintf(azcopyMSIClientID+"="+azureAuthConfig.UserAssignedIdentityID))
+		}
+		klog.V(2).Infof("use system-assigned managed identity to authorize azcopy")
```

Comment:
```
else {
   klog.V(2).Infof("use system-assigned managed identity to authorize azcopy")
}
```
otherwise L810 & L815 would be both printed

Reply:
done